src/Adapter/Cart/CommandHandler/UpdateCartDeliverySettingsHandler.php

```diff
@@ -65,36 +67,18 @@ public function __construct(TranslatorInterface $translator, ConfigurationInterf
     /**
      * {@inheritdoc}
      */
-    public function handle(SetFreeShippingToCartCommand $command): void
+    public function handle(UpdateCartDeliverySettingsCommand $command): void
     {
         $cart = $this->getCart($command->getCartId());
 
-        $backOfficeOrderCode = sprintf('%s%s', CartRule::BO_ORDER_CODE_PREFIX, $cart->id);
-
-        $freeShippingCartRule = $this->getCartRuleForBackOfficeFreeShipping($backOfficeOrderCode);
-
-        if ($command->allowFreeShipping()) {
-            if (null === $freeShippingCartRule) {
-                $freeShippingCartRule = $this->createCartRule($cart, $backOfficeOrderCode);
-            }
-            $cart->addCartRule((int) $freeShippingCartRule->id);
-
-            return;
-        }
-
-        if (null === $freeShippingCartRule) {
-            return;
+        if (!\Validate::isMessage($command->getGiftMessage())) {
```

Comment:
I dont want to do this later because I dont want to start updating the cart then halfway I find an issue and I stop the process.

I want all the validation to happen at the very first, then if all the data is valid I proceed to cart update.

I will fix this by only performing validation if gift message is not null

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/21331/commits/2d6d2a7b4cfda0023ea2cef41026d1bf9c55ed68